### PR TITLE
resolve homedir of sudo user

### PR DESCRIPTION
**Pull request by steiler**

### CODE REVIEW

File: clab/authz_keys.go [hunk 1]
```diff
@@ -18,24 +19,35 @@ import (
...
+		userId = "0" // default to root
+	}
+
+	// lookup user to figure out Home Directory
+	u, err := user.LookupId(userId)
+	if err != nil {
+		return err
+	}
+
+	p := utils.ResolvePath(filepath.Join(u.HomeDir, pubKeysGlobPart), "")
```
Comment (hellt) [inline clab/authz_keys.go:43]: Hi @steiler 
this is a good idea, but I'd rather do this centrally in `utils.ResolvePath`
There you can use the code branch that resolves the `~` char to the user's home directory. You can just use the code block that looks up the SUDO_UID there to identify which user's home dir should it be resolved to

Comment (steiler) [reply clab/authz_keys.go:43]: Alright, yes, that definetely makes sense. Thanks!

File: clab/authz_keys.go [hunk 2]
```diff
@@ -18,24 +18,24 @@ import (
 )
 
 const (
-	pubKeysGlob = "~/.ssh/*.pub"
+	pubKeysGlobPart = "~/.ssh/*.pub"
 	// authorized keys file path on a clab host that is used to create the clabAuthzKeys file.
-	authzKeysFPath = "~/.ssh/authorized_keys"
+	authzKeysFPathPart = "~/.ssh/authorized_keys"
```
Comment (hellt) [inline clab/authz_keys.go:23]: These are not `...Part` anymore I presume

Comment (steiler) [reply clab/authz_keys.go]: sure yes

File: utils/file.go [hunk 3]
```diff
@@ -169,10 +170,24 @@ func ReadFileContent(file string) ([]byte, error) {
...
+		return curUserHomeDir
+	}
+
+	// lookup user to figure out Home Directory
+	u, err := user.LookupId(userId)
+	if err != nil {
+		return curUserHomeDir
+	}
+
+	p = strings.Replace(p, "~", u.HomeDir, 1)
```
Comment (hellt) [inline utils/file.go:190]: In 5c6fe6b I used the current user home dir as a backup dir for cases when sudo user is failed to be retrieved.
If you OK this change we can merge

Comment (steiler) [reply utils/file.go:190]: Yes sure, totally fine with that!

**Pull request merged**